### PR TITLE
[SYCL][HIP] Vendor ID for AMD devices is expected to be 4098

### PR DESCRIPTION
**Pull request by zjin-lcf**
A pull request to solve the issue https://github.com/intel/llvm/issues/6123
Thanks for your review.

### CODE REVIEW

File: sycl/plugins/hip/pi_hip.cpp [hunk 1]
```diff
@@ -980,7 +980,7 @@ pi_result hip_piDeviceGetInfo(pi_device device, pi_device_info param_name,
                    PI_DEVICE_TYPE_GPU);
   }
   case PI_DEVICE_INFO_VENDOR_ID: {
-    return getInfo(param_value_size, param_value, param_value_size_ret, 4318u);
+    return getInfo(param_value_size, param_value, param_value_size_ret, 4098u);
```
Comment (densamoilov) [inline sycl/plugins/hip/pi_hip.cpp:983]: Out of curiosity, what vendor ID is returned when DPC++ was built with HIP NVIDIA support?

Comment (zjin-lcf) [reply sycl/plugins/hip/pi_hip.cpp:983]: Thank you for your suggestion. I don't have a platform for the test, but I added the following:
  
 case PI_DEVICE_INFO_VENDOR_ID: {
#if defined(__HIP_PLATFORM_AMD__)
    pi_uint32 vendor_id = 4098u;
#elif defined(__HIP_PLATFORM_NVIDIA__)
    pi_uint32 vendor_id = 4318u;
#else
    pi_uint32 vendor_id = 0u;
#endif
    return getInfo(param_value_size, param_value, param_value_size_ret, vendor_id);
  }

Comment (steffenlarsen) [reply sycl/plugins/hip/pi_hip.cpp:983]: @zjin-lcf - Are you planning on incorporating the changes you mention here?

Comment (zjin-lcf) [reply sycl/plugins/hip/pi_hip.cpp:983]: I checked the box 'Allow edits and access to secrets by maintainers', so assumed that reviewers could incorporate the changes.

Comment (zjin-lcf) [reply sycl/plugins/hip/pi_hip.cpp]: Okay. I updated the codes for pull request.

**Review by steffenlarsen — DISMISSED** [review]
@AerialMantis | @npmiller - Could you please have a look?

File: sycl/plugins/hip/pi_hip.cpp [hunk 2]
```diff
@@ -980,7 +980,15 @@ pi_result hip_piDeviceGetInfo(pi_device device, pi_device_info param_name,
...
   }
   case PI_DEVICE_INFO_VENDOR_ID: {
-    return getInfo(param_value_size, param_value, param_value_size_ret, 4318u);
+    #if defined(__HIP_PLATFORM_AMD__)
+      pi_uint32 vendor_id = 4098u;
+    #elif defined(__HIP_PLATFORM_NVIDIA__)
+      pi_uint32 vendor_id = 4318u;
+    #else
+      pi_uint32 vendor_id = 0u;
+    #endif
```
Comment (npmiller) [inline sycl/plugins/hip/pi_hip.cpp:989]: The indentation is wrong, this needs to be clang-formatted:
```suggestion
#if defined(__HIP_PLATFORM_AMD__)
    pi_uint32 vendor_id = 4098u;
#elif defined(__HIP_PLATFORM_NVIDIA__)
    pi_uint32 vendor_id = 4318u;
#else
    pi_uint32 vendor_id = 0u;
#endif

```

**Review by npmiller — DISMISSED** [review]

**Review by npmiller — APPROVED** [review]

**Comment by zjin-lcf** [comment]
@npmiller 
Thank you for your comment. How do you format codes automatically ?

**Comment by npmiller** [comment]
> @npmiller Thank you for your comment. How do you format codes automatically ?

You should use `clang-format` for DPC++/LLVM code, there's some pointers about how to use it to format patches here:

* https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#development

And it can also be integrated with IDEs or editors (for example I personally use this plugin to call it directly from vim: https://github.com/frasercrmck/formative.vim )

**Comment by steffenlarsen** [comment]
> How do you format codes automatically ?

You can also use https://github.com/intel/llvm/blob/sycl/clang/tools/clang-format/clang-format-diff.py to apply clang-format from a diff. See the comment at the top of the file. The PR testing should also tell us when things are not formatted in accordance with recent clang-format, though it seems testing did not run for this PR until now.

It should also be possible to create a git hook for it.

**Review by steffenlarsen — APPROVED** [review]

**Comment by densamoilov** [comment]
@zjin-lcf, is there anything else that needs to be done to merge this PR?

**Comment by steffenlarsen** [comment]
Failing tests were disabled in https://github.com/intel/llvm-test-suite/pull/1034. I believe this is ready to be merged.

**Pull request merged**